src/linter/ui5Types/FixHintsGenerator.ts

```diff
@@ -0,0 +1,827 @@
+	}],
+	["measure.getPendingInteractionMeasurement", {
+		moduleName: "sap/ui/performance/trace/Interaction", exportNameToBeUsed: "getPending",
+	}],
+	["fesr.getCurrentTransactionId", {
+		moduleName: "sap/ui/performance/trace/Passport", exportNameToBeUsed: "getCurrentTransactionId",
+	}],
+	["fesr.getRootId", {
+		moduleName: "sap/ui/performance/trace/Passport", exportNameToBeUsed: "getRootId",
+	}],
+	["passport.setActive", {
+		moduleName: "sap/ui/performance/trace/Passport", exportNameToBeUsed: "setActive",
+	}],
+	["passport.traceFlags", {
+		moduleName: "sap/ui/performance/trace/Passport", exportNameToBeUsed: "traceFlags",
+	}],
+	// https://github.com/SAP/ui5-linter/issues/563
+
+	// TODO MB: Possibly missing some replacements, see https://github.com/SAP/ui5-linter/issues/563
+	["act.isActive", {
+		moduleName: "sap/ui/util/ActivityDetection", exportNameToBeUsed: "isActive",
+	}],
+	["initMobile", {
+		moduleName: "sap/ui/util/Mobile", exportNameToBeUsed: "init",
+	}],
+	["setIcons", {
+		moduleName: "sap/ui/util/Mobile", exportNameToBeUsed: "setIcons",
+	}],
+	["setMobileWebAppCapable", {
+		moduleName: "sap/ui/util/Mobile", exportNameToBeUsed: "setWebAppCapable",
+	}],
+	["storage", {
+		moduleName: "sap/ui/util/Storage", exportCodeToBeUsed: "new $moduleIdentifier($1, $2)",
+	}],
+	["getParseError", {
+		moduleName: "sap/ui/util/XMLHelper", exportNameToBeUsed: "getParseError",
+	}],
+	["parseXML", {
+		moduleName: "sap/ui/util/XMLHelper", exportNameToBeUsed: "parse",
+	}],
+	["serializeXML", {
+		moduleName: "sap/ui/util/XMLHelper", exportNameToBeUsed: "serialize",
+	}],
+
+	["device.is.standalone", {
+		exportCodeToBeUsed: "window.navigator.standalone",
+	}],
+	["support.retina", {
+		exportCodeToBeUsed: "window.devicePixelRatio >= 2",
+	}],
+	// TODO: Shall we add validation for a string?
+	["startsWith", {
+		exportCodeToBeUsed: "$1.startsWith($2)",
+	}],
+	// TODO: Shall we add validation for a string?
+	["startsWithIgnoreCase", {
+		exportCodeToBeUsed: "$1.toLowerCase().startsWith($2)",
+	}],
+	// TODO: Shall we add validation for a string?
+	["endsWith", {
+		exportCodeToBeUsed: "$1.endsWith($2)",
+	}],
+	// TODO: Shall we add validation for a string?
+	["endsWithIgnoreCase", {
+		exportCodeToBeUsed: "$1.toLowerCase().endsWith($2)",
+	}],
+	// TODO: Shall we add validation for a string?
+	// jquery guard: jQuery.sap.assert(typeof sPadChar === 'string' && sPadChar)
+	["padLeft", {
+		exportCodeToBeUsed: "$1.padStart($3, $2)",
+	}],
+	// TODO: Shall we add validation for a string?
+	// jquery guard: jQuery.sap.assert(typeof sPadChar === 'string' && sPadChar)
+	["padRight", {
+		exportCodeToBeUsed: "$1.padEnd($3, $2)",
+	}],
+	["delayedCall", {
+		exportCodeToBeUsed: "window.setTimeout($3.bind($2), $1)",
+	}],
+	["clearDelayedCall", {
+		exportCodeToBeUsed: "window.clearTimeout($1)",
+	}],
+	["intervalCall", {
+		exportCodeToBeUsed: "window.setInterval($3.bind($2), $1)",
```

Comment:
I think we mustn't generate a `bind()` here since it would male it impossible to use clearInterval with the original callback variable. Same for `delayedCall`

Reply:
`clearTimeout` / `clearInterval` are not called with the original callback, but with the "handle" that their counterpart `setTimeout` / `setInterval` returns.

Reply:
Fully, agree with that oppinion!
However, the internal implementation of `dellayedCall` and `intervallCall` always executes the method from the correct context: https://github.com/SAP/openui5/blob/master/src/sap.ui.core/src/jquery.sap.script.js#L339. If we don't do that, tests in openui5 will fail: https://github.com/SAP/openui5/blob/master/src/sap.ui.core/test/sap/ui/core/qunit/util/jquery.sap.script.qunit.js#L10

IMO, making it context "free" would be a backwards incompatible change

Reply:
Alright, thanks for the feedback. Then the replacement makes sense 👍